sqlglot/parser.py

```diff
@@ -1105,7 +1109,12 @@ def _parse_join(self):
         if not self._match(TokenType.JOIN):
             return None
 
-        kwargs = {"this": self._parse_table()}
+        this = (
+            self._parse_lateral()
+            if self._curr and self._curr.token_type == TokenType.LATERAL
+            else self._parse_table()
+        )
+        kwargs = {"this": this}
```

Comment:
this could just be {"this": self._parse_lateral() or self._parse_table()} right?

Reply:
Yeah good point